source/expressions.tex

```diff
@@ -4683,6 +4686,9 @@
 the coefficient $\tcode{r}_i$
 of the base-2 representation of the result \tcode{r}
 is 1 if $\tcode{x}_i$ is 0, and 0 otherwise.
+\begin{note}
+The result is the ones' complement (or unary complement) function of the operand.
```

Comment:
I'm actually not sure if "unary complement" on L4695 means "ones' complement", or just "the (unary) "complement"-operator" (where "complement" is not otherwise qualified), given that `~` is also spelled `compl`.

Reply:
I'm pretty sure "unary complement" is just referring to the arity of the operator: the "ones" in "ones' complement" refers to all-bits-set (0xffff), and "unary" would _count_ those bits.  That said, "complement of _x_" here just means "_X_-_x_ for some _X_ determined from context" and is at least [300 years older](https://en.wikipedia.org/wiki/Method_of_complements#Practical_uses) than its use in binary.  Since there is obviously two's complement as well, I think the note should just say
> The result is the ones' complement of the operand.

Reply:
OK -- should we then also modify "unary complement operator" below, since we don't currently define "complement operator"?

Reply:
There isn't any other `~` operator, so I don't think it's harmful the way it is.  That said, it would be fine to just delete "unary complement" there; those two words are already available in the subclause heading and note.  (I would be tempted to start the paragraph with "The operand of the complement operator `~` shall&hellip;", but none of the other descriptions actually _name_ the operators except in the subclause names.)

Reply:
Middle ground: "The result of the `~` operator shal...". That's in line with the other paragraphs, but we don't say "unary" since there is no binary version.

Reply:
Sure (with "operand" instead of "result").

Reply:
Yes, sorry, that's what I actually have in this patch now. PTAL!

Reply:
For what it's worth: [ISO/IEC 2382:2015](https://www.iso.org/obp/ui/#iso:std:iso-iec:2382:ed-1:v2:en), which we normatively cite for definitions of terms such as this, does indeed define "complement" consistent with @opensdh's suggestion:

> ### complement
> number that is derived from a given number by subtracting it from a specified number
>
> Note 1 to entry: Example: In fixed radix notation, the specified number is typically a power of the radix or one less than a given power of the radix.
>
> Note 2 to entry: A negative of a number is often represented by its complement.